core/src/main/java/org/jruby/util/IOChannel.java

```diff
@@ -184,4 +192,10 @@ public int write(ByteBuffer src) throws IOException {
             return write(write, src);
         }
     }
+
+    private class ReadPartialBufferOverflowException extends IOException {
```

Comment:
nitpick: make the class static

Reply:
You're right, I'll fix it.

Reply:
I opted to make it a public, top-level class in our `exceptions` package, since there may also be a need to catch this specific exception some day.